### PR TITLE
add scatter_or_line argument to Learner1D.plot

### PR DESCRIPTION
**Pull request by basnijholt**
I've needed this many times and the discussion in the chat now also made it clear that this is needed.

Now you won't have to look at plots like this: 
image

### CODE REVIEW

**Comment by akhmerov** [comment]
Not sure; we're already using not-a-plotting-library (holoviews), so I'd rather keep adaptive's plotting interface as small as possible.

File: adaptive/learner/learner1D.py [hunk 1]
```diff
@@ -580,22 +580,33 @@ def _loss(self, mapping, ival):
...
-    def plot(self):
+    def plot(self, *, scatter_or_line=None):
         """Returns a plot of the evaluated data.
 
         Returns
         -------
         plot : `holoviews.element.Scatter` (if vdim=1)\
                else `holoviews.element.Path`
             Plot of the evaluated data.
+        scatter_or_line : str, optional
```
Comment (jbweston) [inline adaptive/learner/learner1D.py]: This is a parameter, not a return value

Comment (jbweston) [reply adaptive/learner/learner1D.py]: also set the default value to `scatter` and mention this in the docstring

File: adaptive/learner/learner1D.py [hunk 2]
```diff
@@ -580,22 +580,33 @@ def _loss(self, mapping, ival):
...
+            Plot as a scatter plot ("scatter") or a line plot ("line").
+            By default a line plot will be chosen if the data consists of
+            vectors, otherwise it is a scatter plot.
         """
         hv = ensure_holoviews()
 
         xs, ys = zip(*sorted(self.data.items())) if self.data else ([], [])
         if self.vdim == 1:
-            p = hv.Path([]) * hv.Scatter((xs, ys))
+            if scatter_or_line is None or scatter_or_line == "scatter":
```
Comment (jbweston) [inline adaptive/learner/learner1D.py]: once `scatter_or_line` has a default value of `scatter` you can change this test and the one below to be `if scatter_or_line == "scatter`

File: adaptive/learner/learner1D.py [hunk 3]
```diff
@@ -580,22 +580,33 @@ def _loss(self, mapping, ival):
...
         -------
         plot : `holoviews.element.Scatter` (if vdim=1)\
                else `holoviews.element.Path`
             Plot of the evaluated data.
+        scatter_or_line : str, optional
+            Plot as a scatter plot ("scatter") or a line plot ("line").
+            By default a line plot will be chosen if the data consists of
+            vectors, otherwise it is a scatter plot.
         """
         hv = ensure_holoviews()
```
Comment (jbweston) [inline adaptive/learner/learner1D.py]: add a check here:
```
if scatter_or_line not in ("scatter", "line"):
    raise ValueError("scatter_or_line must be 'scatter' or 'line'")
```

Comment (basnijholt) [reply adaptive/learner/learner1D.py]: Done!

File: adaptive/learner/learner1D.py (hunk 2 above)
Comment (basnijholt) [reply adaptive/learner/learner1D.py]: Simplified it a bit more.

File: adaptive/learner/learner1D.py (hunk 1 above)
Comment (basnijholt) [reply adaptive/learner/learner1D.py]: Done!

**Pull request merged**